### PR TITLE
 [MIG][9.0] hr_period Adapt to date_range

### PR DESCRIPTION
**Pull request by AaronHForgeFlow**
Supersedes this PR https://github.com/OCA/hr/pull/341

Adapted uses the date_range module from server-tools repo. 

### Known issues:
- It's not possible to create date ranges with different number of days using the date_range wizard. That's why periods are created manually for semi-monthly periods.
- The date_range module does not allow to create periods for a single day. This module does. That's why I created this PR: https://github.com/OCA/server-tools/pull/921

### CODE REVIEW

**Comment by MiquelRForgeFlow** [comment]
Please rebase the PR  🙂

**Review by gurneyalex — CHANGES_REQUESTED** [review]
CI is red

**Comment by gurneyalex** [comment]
Odoo PR will be available in 11 only :-/

**Comment by lreficent** [comment]
ping @aheficent

File: hr_period/models/hr_fiscal_year.py [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+    schedule_pay = fields.Selection(
+        get_schedules,
+        'Scheduled Pay',
+        required=True,
+        states={'draft': [('readonly', False)]},
+        default='monthly'
+    )
+    type_id = fields.Many2one(
+        'date.range.type',
+        'Data Range Type',
```
Comment (MiquelRForgeFlow) [inline hr_period/models/hr_fiscal_year.py]: Data -> Date

**Comment by feketemihai** [comment]
@aheficent Do you still need this for v9?

**Comment by AaronHForgeFlow** [comment]
Not really, let's close it. I'll reopen it if anyone is interested.

**Pull request closed without merging**